src/Native/System.Private.CoreLib.Native/pal_threading.cpp

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_common.h"
+#include "pal_threading.h"
+#include "pal_time.h"
+
+#include <sched.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// LowLevelMutex
+
+extern "C" void CoreLibNative_LowLevelMutex_Acquire(LowLevelMutex *mutex)
+{
+    assert(mutex != nullptr);
+    mutex->Acquire();
+}
+
+extern "C" void CoreLibNative_LowLevelMutex_Release(LowLevelMutex *mutex)
+{
+    assert(mutex != nullptr);
+    mutex->Release();
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// LowLevelMonitor
+
+#if !(HAVE_MACH_ABSOLUTE_TIME || HAVE_PTHREAD_CONDATTR_SETCLOCK && HAVE_CLOCK_MONOTONIC)
+#error Don't know how to perfom timed wait on this platform
+#endif
+
+LowLevelMonitor::LowLevelMonitor(bool abortOnFailure, bool *successRef) : LowLevelMutex(abortOnFailure, successRef)
+{
+    if (successRef != nullptr && !*successRef)
+    {
+        assert(!abortOnFailure);
+        return;
+    }
+
+#if HAVE_MACH_ABSOLUTE_TIME
+    // Older versions of OSX don't support CLOCK_MONOTONIC, so we don't use pthread_condattr_setclock. See
+    // Wait(int32_t timeoutMilliseconds).
+    int initError = pthread_cond_init(&m_condition, nullptr);
+#else
+    int error;
+
+    pthread_condattr_t conditionAttributes;
+    error = pthread_condattr_init(&conditionAttributes);
+    if (error != 0)
+    {
+        if (abortOnFailure)
+        {
+            abort();
+        }
+        LowLevelMutex::~LowLevelMutex();
+        *successRef = false;
+        return;
+    }
+
+    error = pthread_condattr_setclock(&conditionAttributes, CLOCK_MONOTONIC);
+    assert(error == 0);
```

Comment:
Yea. Currently in configure.cmake only checking for the existence of CLOCK_MONOTONIC and pthread_condattr_setclock, I think that should be sufficient because CLOCK_MONOTONIC is the typical argument passed to that function to change the clock from the default.